File: src/main/java/org/antop/java/chapter2/Rectangle.java
package org.antop.java.chapter2;

public class Rectangle {
    private final int height;
    private final int weight;

    public Rectangle(int height, int weight) {
        this.height = height;
        this.weight = weight;
    }

    public int getHeight() {
        return height;
    }

    public int getWeight() {
        return weight;
    }

    public boolean isSquare() {
        return height == weight;
    }

    public static void main(String[] args) {
        Rectangle r = new Rectangle(200, 200);
        r.isSquare(); // true
    }

}

File: src/main/kotlin/org/antop/kotlin/chapter2/Rectangle.kt
package org.antop.kotlin.chapter2

class Rectangle(val height: Int, val width: Int) {
    val isSquare: Boolean
        get() { // 프로퍼티 게터 선언
            return height == width
        }
}

fun main() {
    val rectangle = Rectangle(100, 100)
    rectangle.isSquare // true
}
